### ISSUE TITLE
Problems with widgetVars in PrimeFaces 14 (unrendered and hyphens)

### ISSUE THREAD

**Issue by clx-cruis** [primefaces/primefaces#12765]
### Describe the bug

Hi!

We wanted to upgrade Primefaces in a project and came across two issues with widgetVars. This didn't happen in PrimeFaces 13 but only in Primefaces 14 which prevents us from upgrading to 14.

1. Problem:  In case of using update="@widgetVar(testInputText)" and the component with the widgetVar="testInputText" is not rendered we are facing this error: "Cannot find component for expressions "@widgetVar(testInputText)"" when the page is loaded. When using dynamic rendering conditions it is difficult to handle this. See reproducer [primefaces-test-master_UnrenderedProblem.zip](https://github.com/user-attachments/files/17491064/primefaces-test-master_UnrenderedProblem.zip)

2. Problem: The pattern for widgetVar matching doesn't allow hyphen "-" to be used in widgetVars anymore. It results in error: "Expression does not match following pattern @widgetVar(var). Expression: "widgetVar(test-Panel)"". Is there any reason for not allowing hyphens in widgetVars? Or could this regex pattern in WidgetVarSearchKeywordResolver maybe be changed in futures release to include hyphens? 
![image](https://github.com/user-attachments/assets/9d04e3ca-ee8c-4354-a5e9-dd361d69d841)
Also attached reproducer for this, see [primefaces-test-master_hyphen.zip](https://github.com/user-attachments/files/17491067/primefaces-test-master_hyphen.zip)

Thanks a lot for your work! Really appreciate it.

Best regards

Christian

### PrimeFaces edition

Community

### PrimeFaces version

14.0.6

### JSF implementation

Mojarra

### JSF version

4.0

### Java version

21

**Comment by melloware** [comment]
Similar to:https://github.com/primefaces/primefaces/issues/12701

**Comment by tandraschko** [comment]
`-` was not allowed because of the legacy namespace and older browsers
now it propably doesnt matter anymore

please also check Widget#resolveWidgetVar

**Comment by tandraschko** [comment]
and please create a seperate issue for #1
mixing multiple issues in one is only chaos

**Comment by melloware** [comment]
@tandraschko issue 1 is already https://github.com/primefaces/primefaces/issues/12701

**Comment by tandraschko** [comment]
Issue 1 should be easy:

```
    public AjaxRequestBuilder update(UIComponent component, String expressions, boolean ignoreNoResult) {
        addExpressions(component, expressions, "u", ignoreNoResult ? HINTS_UPDATE_IGNORE_NO_RESULT : HINTS_UPDATE, true);
        return this;
    }
```

can you change true to false as last param and also check if it fixes #12701?

**Comment by melloware** [comment]
yep let me try.

**Comment by melloware** [comment]
It does **NOT** fix 12701

**Comment by tandraschko** [comment]
But it should fix issue 1 here

**Comment by melloware** [comment]
let me test i used the reproducer from 12701

**Comment by melloware** [comment]
OK using the reproducer above and setting to `false` i still get.

```
SEVERE: Error Rendering View[/test.xhtml]
jakarta.faces.component.search.ComponentNotFoundException: Cannot find component for expressions "@widgetVar(testInputText)" referenced from "frmTest:j_idt8".
        at com.sun.faces.component.search.SearchExpressionHandlerImpl.resolveClientIds(SearchExpressionHandlerImpl.java:116)
        at org.primefaces.expression.SearchExpressionUtils.resolveClientIdsAsString(SearchExpressionUtils.java:134)
        at org.primefaces.util.AjaxRequestBuilder.addExpressions(AjaxRequestBuilder.java:162)
        at org.primefaces.util.AjaxRequestBuilder.update(AjaxRequestBuilder.java:154)
        at org.primefaces.renderkit.CoreRenderer.preConfiguredAjaxRequestBuilder(CoreRenderer.java:487)
        at org.primefaces.renderkit.CoreRenderer.buildAjaxRequest(CoreRenderer.java:508)
        at org.primefaces.renderkit.CoreRenderer.buildAjaxRequest(CoreRenderer.java:504)
        at org.primefaces.component.commandbutton.CommandButtonRenderer.buildRequest(CommandButtonRenderer.java:155)
        at org.primefaces.component.commandbutton.CommandButtonRenderer.encodeMarkup(CommandButtonRenderer.java:84)
        at org.primefaces.component.commandbutton.CommandButtonRenderer.encodeEnd(CommandButtonRenderer.java:68)
        at jakarta.faces.component.UIComponentBase.encodeEnd(UIComponentBase.java:585)
        at jakarta.faces.component.UIComponent.encodeAll(UIComponent.java:1442)
        at jakarta.faces.render.Renderer.encodeChildren(Renderer.java:146)
        at jakarta.faces.component.UIComponentBase.encodeChildren(UIComponentBase.java:556)
        at jakarta.faces.component.UIComponent.encodeAll(UIComponent.java:1435)
        at jakarta.faces.component.UIComponent.encodeAll(UIComponent.java:1438)
        at jakarta.faces.component.UIComponent.encodeAll(UIComponent.java:1438)
        at com.sun.faces.application.view.FaceletViewHandlingStrategy.renderView(FaceletViewHandlingStrategy.java:449)
```

**Comment by tandraschko** [comment]
this will fix #1
